File: MyAccount.vb
Imports System.ComponentModel
Imports System.Drawing.Drawing2D
Imports MySql.Data.MySqlClient

Public Class MyAccount
    Dim b1 As String
    Dim a As String
    Dim b As String
    Dim c As String
    Dim setUname As String
    Dim index1 As Integer
    Dim index3 As Integer
    Dim flag As Integer

    Public Property passEmail As String
    Public Property passUsername As String

    ReadOnly opf As New OpenFileDialog With {
        .Filter = "Choose Image(*.JPG;*.PNG;*.GIF)|*.jpg;*.png;*.gif"
    }

    Private Sub MyAccount_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If
        conn.ConnectionString = "server=localhost;user id=root;port=3306;password=0102bryan;database=accountdb"

        If My.Settings.userRememberMe = True Then
            passEmail = My.Settings.userEmail
            passUsername = My.Settings.userUsername
        End If

        getProfilePic()
        setCredentials()
        userNameChangeChecker()

        Dim outputEmail As Char
        outputEmail = passEmail.Substring(0, 1)
        Dim outputEmail1 As Char
        outputEmail1 = passEmail.Substring(1, 1)
        Dim count As Integer
        count = passEmail.Length

        Dim split = passEmail.Split("@"c)
        If (split.Count = 2) Then
            b1 = split(1).ToString
        End If

        If count > 18 Then
            lblEmail.Text = outputEmail & outputEmail1 & "***********@" & b1
        Else
            lblEmail.Text = outputEmail & outputEmail1 & "******@" & b1
        End If

        txtPhoneNum.MaxLength = 11
    End Sub

    Private Sub userNameChangeChecker()
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If

        Try
            sql = "Select * FROM userProfile WHERE uEmail = '" & passEmail & "'"
            connect()

            If dr.Read Then
                Dim count As String
                count = dr.GetInt32(2).ToString
                If count = "0" Then
                    txtUsername.Enabled = True
                Else
                    txtUsername.Enabled = False
                    lblUNInfo.Text = "You already changed your username once."
                End If
            End If
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
        conn.Close()
    End Sub

    Private Sub setCredentials()
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If

        Try
            sql = "SELECT * FROM userProfile WHERE uEmail = '" & passEmail & "'"
            connect()

            If dr.Read Then
                setUname = dr.GetString("uName")
                txtUsername.Text = setUname
                txtFullName.Text = dr.GetString("userName")
                txtPhoneNum.Text = dr.GetString("userPhone")
                If Not dr.IsDBNull(8) Then
                    If dr.GetString("userGend").Equals("Male") Then
                        btnMale.Checked = True
                    ElseIf dr.GetString("userGend").Equals("Female") Then
                        btnFemale.Checked = True
                    ElseIf dr.GetString("userGend").Equals("Other") Then
                        btnOther.Checked = True
                    Else
                        btnMale.Checked = False
                        btnFemale.Checked = False
                        btnOther.Checked = False
                    End If
                Else
                    btnMale.Checked = False
                    btnFemale.Checked = False
                    btnOther.Checked = False
                End If
                If Not dr.IsDBNull(9) Then
                    lblDate.Text = dr.GetDateTime("userBday")
                    Dim split = lblDate.Text.Split("/"c)
                    If (split.Count = 3) Then
                        a = split(0)
                        b = split(1)
                        c = split(2)
                    End If
                    index1 = cmbYear.FindString(c)
                    cmbYear.SelectedIndex = index1
                    cmbMonth.SelectedIndex = a - 1
                    index3 = cmbDate.FindString(b)
                    cmbDate.SelectedIndex = index3
                End If
            End If
        Catch ex As Exception
            conn.Close()
        End Try
        conn.Close()

        btnSave.Visible = False
        btnDiscard.Visible = False
    End Sub

    Private Sub saveCredentials()
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If

        Try
            sql = "SELECT * FROM userProfile WHERE uEmail = '" & passEmail & "'"
            connect()

            If dr.Read Then
                Dim uname = dr.GetString("uName")

                If dr.GetInt32(2) = "0" Then
                    If unameChecker(txtUsername.Text) Then
                        lblUsernameChecker.Visible = True
                        btnSave.Enabled = True
                        btnDiscard.Enabled = True
                    Else
                        If Not txtUsername.Text.Equals(uname) Then
                            If conn.State = ConnectionState.Open Then
                                conn.Close()
                            End If

                            sql = "UPDATE userProfile SET uName = '" & txtUsername.Text & "', uNCount = '" & "1" & "' WHERE uEmail = '" & passEmail & "'"
                            connect()

                            dr.Close()

                            If cmd.ExecuteNonQuery Then
                                If lblUsernameChecker.Visible = False Then
                                    SavedConfirmation.ShowDialog()
                                End If
                            End If
                        End If
                    End If
                End If

                If conn.State = ConnectionState.Open Then
                    conn.Close()
                End If

                Dim gender As String
                If btnMale.Checked = True Then
                    gender = "Male"
                ElseIf btnFemale.Checked = True Then
                    gender = "Female"
                Else
                    gender = "Other"
                End If

                Dim bday As String
                Dim a = cmbYear.Text
                Dim b = cmbMonth.SelectedIndex + 1
                Dim c = cmbDate.Text
                bday = a & "-" & b & "-" & c

                sql = "UPDATE userProfile SET userName = '" & txtFullName.Text & "', userPhone = '" & txtPhoneNum.Text & "', userGend = '" & gender & "', userBday = '" & bday & "' WHERE uEmail = '" & passEmail & "'"
                connect()

                dr.Close()

                If cmd.ExecuteNonQuery Then
                    If Not flag = 1 Then
                        If lblUsernameChecker.Visible = False Then
                            SavedConfirmation.ShowDialog()
                        End If
                    End If
                End If
            End If
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
        conn.Close()
    End Sub

    Private Function unameChecker(ByVal userName) As Boolean
        Dim usernameExist As Boolean = False

        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If
        Try
            sql = "SELECT * FROM userProfile WHERE uName = '" & userName & "'"
            connect()

            If dr.Read Then
                Dim email = dr.GetString("uEmail")
                If Not email.Equals(passEmail) Then
                    usernameExist = True
                End If
            End If
        Catch ex As Exception
            MsgBox(ex.Message)
        End Try
        conn.Close()
        Return usernameExist
    End Function

    Private Sub getProfilePic()
        If conn.State = ConnectionState.Open Then
            conn.Close()
        End If

        Dim arrimage() As Byte

        Try
            sql = "SELECT userPic FROM userProfile WHERE uEmail = '" & passEmail & "'"
            connect()

            If dr.Read Then
                arrimage = dr.Item("userPic")
                Dim mstream As New System.IO.MemoryStream(arrimage)
                setPicture.CustomImages.Image = System.Drawing.Image.FromStream(mstream)

                Dim originalImage = setPicture.CustomImages.Image
                Dim croppedImage As New Bitmap(originalImage.Width, originalImage.Height)

                Using g = Graphics.FromImage(croppedImage)
                    Dim path As New GraphicsPath

                    path.AddEllipse(0, 0, croppedImage.Width, croppedImage.Height)

                    Dim reg As New Region(path)

                    g.Clip = reg
                    g.DrawImage(originalImage, Point.Empty)
                End Using

                setPicture.CustomImages.Image = croppedImage
            ElseIf dr.IsDBNull(5) Then
                conn.Close()
            End If
        Catch ex As Exception
            conn.Close()
        End Try
        conn.Close()
    End Sub

    Private Sub setProfilePic()
        If opf.ShowDialog = Windows.Forms.DialogResult.OK Then
            flag = 1
            Dim originalImage = Image.FromFile(opf.FileName)
            Dim croppedImage As New Bitmap(originalImage.Width, originalImage.Height)

            Using g = Graphics.FromImage(croppedImage)
                Dim path As New GraphicsPath

                path.AddEllipse(0, 0, croppedImage.Width, croppedImage.Height)

                Dim reg As New Region(path)

                g.Clip = reg
                g.DrawImage(originalImage, Point.Empty)
            End Using

            setPicture.CustomImages.Image = croppedImage

            btnSave.Visible = True
            btnDiscard.Visible = True
        End If
    End Sub

    Private Sub saveProfilePic()
        If flag = 1 Then
            Dim originalImage = Image.FromFile(opf.FileName)

            Dim filesize As UInt32

            Dim mstream As New System.IO.MemoryStream
            originalImage.Save(mstream, System.Drawing.Imaging.ImageFormat.Jpeg)
            Dim arrimage() As Byte = mstream.GetBuffer
            filesize = mstream.Length
            mstream.Close()

            If conn.State = ConnectionState.Open Then
                conn.Close()
            End If

            Try
                sql = "UPDATE userProfile SET userPic = @pic WHERE uEmail = @email"
                With cmd
                    .Parameters.Clear()
                    .Parameters.Add("@pic", MySqlDbType.Blob).Value = arrimage
                    .Parameters.Add("@email", MySqlDbType.VarChar).Value = passEmail
                End With
                connect()
                'SavedConfirmation.ShowDialog()
            Catch ex As Exception
                MsgBox(ex.Message)
            End Try
            conn.Close()

            flag = 0
        End If
    End Sub

    Private Sub setPicture_Click(sender As Object, e As EventArgs) Handles setPicture.Click
        setProfilePic()
    End Sub

    Private Sub btnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        Dim obj As New Settings
        obj.getProfilePic()

        saveProfilePic()
        If txtUsername.Text.Contains(" ") Then
            lblUsernameChecker.Visible = True
            lblUsernameChecker.Text = "Username should not contain spaces"
        Else
            saveCredentials()
        End If
        userNameChangeChecker()

        If lblUsernameChecker.Visible = True Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        Else
            btnSave.Visible = False
            btnDiscard.Visible = False
        End If
    End Sub

    Private Sub btnDiscard_Click(sender As Object, e As EventArgs) Handles btnDiscard.Click
        getProfilePic()

        btnDiscard.Visible = False
        btnSave.Visible = False

        setCredentials()
    End Sub

    Private Sub btnSelectPic_Click(sender As Object, e As EventArgs) Handles btnSelectPic.Click
        setProfilePic()
    End Sub

    Private Sub txtUsername_TextChanged(sender As Object, e As EventArgs) Handles txtUsername.TextChanged
        btnSave.Visible = True
        btnDiscard.Visible = True

        lblUsernameChecker.Visible = False
    End Sub

    Private Sub txtFullName_TextChanged(sender As Object, e As EventArgs) Handles txtFullName.TextChanged
        btnSave.Visible = True
        btnDiscard.Visible = True
    End Sub

    Private Sub txtPhoneNum_TextChanged(sender As Object, e As EventArgs) Handles txtPhoneNum.TextChanged
        btnSave.Visible = True
        btnDiscard.Visible = True
    End Sub

    Private Sub btnFemale_CheckedChanged(sender As Object, e As EventArgs) Handles btnFemale.CheckedChanged
        If Not btnFemale.Checked = True Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        ElseIf btnMale.Checked = False And btnOther.Checked = False Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        End If
    End Sub

    Private Sub btnMale_CheckedChanged(sender As Object, e As EventArgs) Handles btnMale.CheckedChanged
        If Not btnMale.Checked = True Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        ElseIf btnFemale.Checked = False And btnOther.Checked = False Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        End If
    End Sub

    Private Sub btnOther_CheckedChanged(sender As Object, e As EventArgs) Handles btnOther.CheckedChanged
        If Not btnOther.Checked = True Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        ElseIf btnMale.Checked = False And btnFemale.Checked = False Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        End If
    End Sub

    Private Sub cmbDate_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cmbDate.SelectedIndexChanged
        If Not cmbDate.SelectedIndex = index3 Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        End If
    End Sub

    Private Sub cmbMonth_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cmbMonth.SelectedIndexChanged
        If Not cmbMonth.SelectedIndex = a - 1 Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        End If
    End Sub

    Private Sub cmbYear_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cmbYear.SelectedIndexChanged
        If Not cmbYear.SelectedIndex = index1 Then
            btnSave.Visible = True
            btnDiscard.Visible = True
        End If
    End Sub

    Private Sub txtPhoneNum_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txtPhoneNum.KeyPress
        If Not Char.IsNumber(e.KeyChar) AndAlso Not Char.IsControl(e.KeyChar) Then e.KeyChar = ""
    End Sub
End Class

File: MyAccount.cs
using System;
using System.Data;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Linq;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using MySql.Data.MySqlClient;

namespace MilkTeaOrderAndInventorySystem
{

    public partial class MyAccount
    {
        private string b1;
        private string a;
        private string b;
        private string c;
        private string setUname;
        private int index1;
        private int index3;
        private int flag;

        public string passEmail { get; set; }
        public string passUsername { get; set; }

        private readonly OpenFileDialog opf = new OpenFileDialog() { Filter = "Choose Image(*.JPG;*.PNG;*.GIF)|*.jpg;*.png;*.gif" };

        public MyAccount()
        {
            InitializeComponent();
        }

        private void MyAccount_Load(object sender, EventArgs e)
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }
            UniversalVariables.conn.ConnectionString = "server=localhost;user id=root;port=3306;password=0102bryan;database=accountdb";

            if (My.MySettingsProperty.Settings.userRememberMe == true)
            {
                passEmail = My.MySettingsProperty.Settings.userEmail;
                passUsername = My.MySettingsProperty.Settings.userUsername;
            }

            getProfilePic();
            setCredentials();
            userNameChangeChecker();

            char outputEmail;
            outputEmail = Conversions.ToChar(passEmail.Substring(0, 1));
            char outputEmail1;
            outputEmail1 = Conversions.ToChar(passEmail.Substring(1, 1));
            int count;
            count = passEmail.Length;

            string[] split = passEmail.Split('@');
            if (split.Count() == 2)
            {
                b1 = split[1].ToString();
            }

            if (count > 18)
            {
                lblEmail.Text = Conversions.ToString(outputEmail) + outputEmail1 + "***********@" + b1;
            }
            else
            {
                lblEmail.Text = Conversions.ToString(outputEmail) + outputEmail1 + "******@" + b1;
            }

            txtPhoneNum.MaxLength = 11;
        }

        private void userNameChangeChecker()
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }

            try
            {
                UniversalVariables.sql = "Select * FROM userProfile WHERE uEmail = '" + passEmail + "'";
                UniversalVariables.connect();

                if (UniversalVariables.dr.Read())
                {
                    string count;
                    count = UniversalVariables.dr.GetInt32(2).ToString();
                    if (count == "0")
                    {
                        txtUsername.Enabled = true;
                    }
                    else
                    {
                        txtUsername.Enabled = false;
                        lblUNInfo.Text = "You already changed your username once.";
                    }
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
            }
            UniversalVariables.conn.Close();
        }

        private void setCredentials()
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }

            try
            {
                UniversalVariables.sql = "SELECT * FROM userProfile WHERE uEmail = '" + passEmail + "'";
                UniversalVariables.connect();

                if (UniversalVariables.dr.Read())
                {
                    setUname = UniversalVariables.dr.GetString("uName");
                    txtUsername.Text = setUname;
                    txtFullName.Text = UniversalVariables.dr.GetString("userName");
                    txtPhoneNum.Text = UniversalVariables.dr.GetString("userPhone");
                    if (!UniversalVariables.dr.IsDBNull(8))
                    {
                        if (UniversalVariables.dr.GetString("userGend").Equals("Male"))
                        {
                            btnMale.Checked = true;
                        }
                        else if (UniversalVariables.dr.GetString("userGend").Equals("Female"))
                        {
                            btnFemale.Checked = true;
                        }
                        else if (UniversalVariables.dr.GetString("userGend").Equals("Other"))
                        {
                            btnOther.Checked = true;
                        }
                        else
                        {
                            btnMale.Checked = false;
                            btnFemale.Checked = false;
                            btnOther.Checked = false;
                        }
                    }
                    else
                    {
                        btnMale.Checked = false;
                        btnFemale.Checked = false;
                        btnOther.Checked = false;
                    }
                    if (!UniversalVariables.dr.IsDBNull(9))
                    {
                        lblDate.Text = Conversions.ToString(UniversalVariables.dr.GetDateTime("userBday"));
                        string[] split = lblDate.Text.Split('/');
                        if (split.Count() == 3)
                        {
                            a = split[0];
                            b = split[1];
                            c = split[2];
                        }
                        index1 = cmbYear.FindString(c);
                        cmbYear.SelectedIndex = index1;
                        cmbMonth.SelectedIndex = (int)Math.Round(Conversions.ToDouble(a) - 1d);
                        index3 = cmbDate.FindString(b);
                        cmbDate.SelectedIndex = index3;
                    }
                }
            }
            catch (Exception ex)
            {
                UniversalVariables.conn.Close();
            }
            UniversalVariables.conn.Close();

            btnSave.Visible = false;
            btnDiscard.Visible = false;
        }

        private void saveCredentials()
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }

            try
            {
                UniversalVariables.sql = "SELECT * FROM userProfile WHERE uEmail = '" + passEmail + "'";
                UniversalVariables.connect();

                if (UniversalVariables.dr.Read())
                {
                    string uname = UniversalVariables.dr.GetString("uName");

                    if (UniversalVariables.dr.GetInt32(2) == Conversions.ToDouble("0"))
                    {
                        if (unameChecker(txtUsername.Text))
                        {
                            lblUsernameChecker.Visible = true;
                            btnSave.Enabled = true;
                            btnDiscard.Enabled = true;
                        }
                        else if (!txtUsername.Text.Equals(uname))
                        {
                            if (UniversalVariables.conn.State == ConnectionState.Open)
                            {
                                UniversalVariables.conn.Close();
                            }

                            UniversalVariables.sql = "UPDATE userProfile SET uName = '" + txtUsername.Text + "', uNCount = '" + "1" + "' WHERE uEmail = '" + passEmail + "'";
                            UniversalVariables.connect();

                            UniversalVariables.dr.Close();

                            if (Conversions.ToBoolean(UniversalVariables.cmd.ExecuteNonQuery()))
                            {
                                if (lblUsernameChecker.Visible == false)
                                {
                                    My.MyProject.Forms.SavedConfirmation.ShowDialog();
                                }
                            }
                        }
                    }

                    if (UniversalVariables.conn.State == ConnectionState.Open)
                    {
                        UniversalVariables.conn.Close();
                    }

                    string gender;
                    if (btnMale.Checked == true)
                    {
                        gender = "Male";
                    }
                    else if (btnFemale.Checked == true)
                    {
                        gender = "Female";
                    }
                    else
                    {
                        gender = "Other";
                    }

                    string bday;
                    string a = cmbYear.Text;
                    int b = cmbMonth.SelectedIndex + 1;
                    string c = cmbDate.Text;
                    bday = a + "-" + b + "-" + c;

                    UniversalVariables.sql = "UPDATE userProfile SET userName = '" + txtFullName.Text + "', userPhone = '" + txtPhoneNum.Text + "', userGend = '" + gender + "', userBday = '" + bday + "' WHERE uEmail = '" + passEmail + "'";
                    UniversalVariables.connect();

                    UniversalVariables.dr.Close();

                    if (Conversions.ToBoolean(UniversalVariables.cmd.ExecuteNonQuery()))
                    {
                        if (!(flag == 1))
                        {
                            if (lblUsernameChecker.Visible == false)
                            {
                                My.MyProject.Forms.SavedConfirmation.ShowDialog();
                            }
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
            }
            UniversalVariables.conn.Close();
        }

        private bool unameChecker(object userName)
        {
            bool usernameExist = false;

            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }
            try
            {
                UniversalVariables.sql = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject("SELECT * FROM userProfile WHERE uName = '", userName), "'"));
                UniversalVariables.connect();

                if (UniversalVariables.dr.Read())
                {
                    string email = UniversalVariables.dr.GetString("uEmail");
                    if (!email.Equals(passEmail))
                    {
                        usernameExist = true;
                    }
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
            }
            UniversalVariables.conn.Close();
            return usernameExist;
        }

        private void getProfilePic()
        {
            if (UniversalVariables.conn.State == ConnectionState.Open)
            {
                UniversalVariables.conn.Close();
            }

            byte[] arrimage;

            try
            {
                UniversalVariables.sql = "SELECT userPic FROM userProfile WHERE uEmail = '" + passEmail + "'";
                UniversalVariables.connect();

                if (UniversalVariables.dr.Read())
                {
                    arrimage = (byte[])UniversalVariables.dr["userPic"];
                    var mstream = new System.IO.MemoryStream(arrimage);
                    setPicture.CustomImages.Image = Image.FromStream(mstream);

                    var originalImage = setPicture.CustomImages.Image;
                    var croppedImage = new Bitmap(originalImage.Width, originalImage.Height);

                    using (var g = Graphics.FromImage(croppedImage))
                    {
                        var path = new GraphicsPath();

                        path.AddEllipse(0, 0, croppedImage.Width, croppedImage.Height);

                        var reg = new Region(path);

                        g.Clip = reg;
                        g.DrawImage(originalImage, Point.Empty);
                    }

                    setPicture.CustomImages.Image = croppedImage;
                }
                else if (UniversalVariables.dr.IsDBNull(5))
                {
                    UniversalVariables.conn.Close();
                }
            }
            catch (Exception ex)
            {
                UniversalVariables.conn.Close();
            }
            UniversalVariables.conn.Close();
        }

        private void setProfilePic()
        {
            if (opf.ShowDialog() == DialogResult.OK)
            {
                flag = 1;
                var originalImage = Image.FromFile(opf.FileName);
                var croppedImage = new Bitmap(originalImage.Width, originalImage.Height);

                using (var g = Graphics.FromImage(croppedImage))
                {
                    var path = new GraphicsPath();

                    path.AddEllipse(0, 0, croppedImage.Width, croppedImage.Height);

                    var reg = new Region(path);

                    g.Clip = reg;
                    g.DrawImage(originalImage, Point.Empty);
                }

                setPicture.CustomImages.Image = croppedImage;

                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
        }

        private void saveProfilePic()
        {
            if (flag == 1)
            {
                var originalImage = Image.FromFile(opf.FileName);

                uint filesize;

                var mstream = new System.IO.MemoryStream();
                originalImage.Save(mstream, System.Drawing.Imaging.ImageFormat.Jpeg);
                byte[] arrimage = mstream.GetBuffer();
                filesize = (uint)mstream.Length;
                mstream.Close();

                if (UniversalVariables.conn.State == ConnectionState.Open)
                {
                    UniversalVariables.conn.Close();
                }

                try
                {
                    UniversalVariables.sql = "UPDATE userProfile SET userPic = @pic WHERE uEmail = @email";
                    {
                        ref var withBlock = ref UniversalVariables.cmd;
                        withBlock.Parameters.Clear();
                        withBlock.Parameters.Add("@pic", MySqlDbType.Blob).Value = arrimage;
                        withBlock.Parameters.Add("@email", MySqlDbType.VarChar).Value = passEmail;
                    }
                    UniversalVariables.connect();
                }
                // SavedConfirmation.ShowDialog()
                catch (Exception ex)
                {
                    Interaction.MsgBox(ex.Message);
                }
                UniversalVariables.conn.Close();

                flag = 0;
            }
        }

        private void setPicture_Click(object sender, EventArgs e)
        {
            setProfilePic();
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            var obj = new Settings();
            obj.getProfilePic();

            saveProfilePic();
            if (txtUsername.Text.Contains(" "))
            {
                lblUsernameChecker.Visible = true;
                lblUsernameChecker.Text = "Username should not contain spaces";
            }
            else
            {
                saveCredentials();
            }
            userNameChangeChecker();

            if (lblUsernameChecker.Visible == true)
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
            else
            {
                btnSave.Visible = false;
                btnDiscard.Visible = false;
            }
        }

        private void btnDiscard_Click(object sender, EventArgs e)
        {
            getProfilePic();

            btnDiscard.Visible = false;
            btnSave.Visible = false;

            setCredentials();
        }

        private void btnSelectPic_Click(object sender, EventArgs e)
        {
            setProfilePic();
        }

        private void txtUsername_TextChanged(object sender, EventArgs e)
        {
            btnSave.Visible = true;
            btnDiscard.Visible = true;

            lblUsernameChecker.Visible = false;
        }

        private void txtFullName_TextChanged(object sender, EventArgs e)
        {
            btnSave.Visible = true;
            btnDiscard.Visible = true;
        }

        private void txtPhoneNum_TextChanged(object sender, EventArgs e)
        {
            btnSave.Visible = true;
            btnDiscard.Visible = true;
        }

        private void btnFemale_CheckedChanged(object sender, EventArgs e)
        {
            if (!(btnFemale.Checked == true))
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
            else if (btnMale.Checked == false & btnOther.Checked == false)
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
        }

        private void btnMale_CheckedChanged(object sender, EventArgs e)
        {
            if (!(btnMale.Checked == true))
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
            else if (btnFemale.Checked == false & btnOther.Checked == false)
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
        }

        private void btnOther_CheckedChanged(object sender, EventArgs e)
        {
            if (!(btnOther.Checked == true))
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
            else if (btnMale.Checked == false & btnFemale.Checked == false)
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
        }

        private void cmbDate_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (!(cmbDate.SelectedIndex == index3))
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
        }

        private void cmbMonth_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (!(cmbMonth.SelectedIndex == Conversions.ToDouble(a) - 1d))
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
        }

        private void cmbYear_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (!(cmbYear.SelectedIndex == index1))
            {
                btnSave.Visible = true;
                btnDiscard.Visible = true;
            }
        }

        private void txtPhoneNum_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!char.IsNumber(e.KeyChar) && !char.IsControl(e.KeyChar))
                e.KeyChar = Conversions.ToChar("");
        }
    }
}